Classes/Doorbell.m

```diff
@@ -170,31 +185,83 @@ - (void)sendSubmit:(NSString*)message email:(NSString*)email
     [submitRequest setHTTPMethod:@"POST"];
-    [submitRequest addValue:UserAgent forHTTPHeaderField:@"User-Agent"];
-    [submitRequest addValue:@"application/json" forHTTPHeaderField:@"Content-Type"];
-
-    NSString *postString = [[NSString alloc] initWithData:jsonData encoding:NSUTF8StringEncoding];
 
-    [submitRequest setHTTPBody:[postString dataUsingEncoding:NSUTF8StringEncoding]];
+    NSURLSessionUploadTask *submitTask = [_session uploadTaskWithRequest:submitRequest
+                                                               fromData:jsonData
+                                                      completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+                                                           if ([response isKindOfClass:[NSHTTPURLResponse class]]) {
+                                                               NSHTTPURLResponse *httpResp = (id)response;
+                                                               NSString *content = [NSString stringWithUTF8String:data.bytes];
+                                                               NSLog(@"%d:%@", (int)httpResp.statusCode, content);
+                                                               dispatch_async(dispatch_get_main_queue(), ^{
+                                                                   [self manageSubmitResponse:httpResp content:content];
+                                                               });
+                                                           }
+                                                       }];
+    [submitTask resume];
+
+    if (self.images.count > 0) {
+        [self uploadImages];
+    }
+}
 
-    [NSURLConnection sendAsynchronousRequest:submitRequest
-                                       queue:[NSOperationQueue mainQueue]
-                           completionHandler:^(NSURLResponse *r, NSData *d, NSError *e) {
-                               if ([r isKindOfClass:[NSHTTPURLResponse class]]) {
-                                   NSHTTPURLResponse *httpResp = (id)r;
-                                   NSString *content = [NSString stringWithUTF8String:d.bytes];
-                                   NSLog(@"%d:%@", (int)httpResp.statusCode, content);
+- (void)uploadImages {
+    NSURLSessionConfiguration *conf = [[NSURLSessionConfiguration alloc] init];
+    [conf setHTTPAdditionalHeaders:@{@"Accept"        : @"application/json",
+                                     @"Content-Type"  : [NSString stringWithFormat:@"multipart/form-data; boundary=%@", self.imageBoundary]
+                                     }];
+    NSURLSession *uploadSession = [NSURLSession sessionWithConfiguration:conf];
 
-                                   [self manageSubmitResponse:httpResp content:content];
-                               }
-                           }];
+    NSString *query = [NSString stringWithFormat:EndpointTemplate, self.appID, @"submit", self.apiKey];
```

Comment:
Yeah I need to change that bit up.  Just wanted to get the structure of it there.